sql/tables.go

```diff
@@ -418,6 +419,16 @@ type TableNode interface {
 	UnderlyingTable() Table
 }
 
+// TableIdNode is a distinct source of rows associated with a table
+// identifier and set of column identifiers.
+type TableIdNode interface {
```

Comment:
Like I said in other places: this interface should be in `plan` or `analyzer` or `memo`, not here

Reply:
makes sense

Reply:
actually I need to do the `TableFunction` refactor first, memory causes a cycle